### ISSUE TITLE
Realm 2.0.0 Update FATAL EXCEPTION : Unable to instantiate receiver io.realm.internal.network.NetworkStateReceiver

### ISSUE THREAD

**Issue by ahmadalibaloch** [realm/realm-java#3521]
After updating to Realm 2.0.0 today this exception occurred on fresh installation (removed old app data for migration problems).

**Update**
**After restarting the app again and again. problem vanished.**

```
 E/AndroidRuntime: FATAL EXCEPTION: main Process: com.xyz.mobileapp, PID: 5829
 java.lang.RuntimeException: Unable to instantiate receiver io.realm.internal.network.NetworkStateReceiver: java.lang.ClassNotFoundException: Didn't find class "io.realm.internal.network.NetworkStateReceiver" on path: DexPathList[[zip file "/data/app/com.xyz.mobileapp-2/base.apk"],nativeLibraryDirectories=[/data/app/com.xyz.mobileapp-2/lib/arm, /vendor/lib, /system/lib]]
                                                                           at android.app.ActivityThread.handleReceiver(ActivityThread.java:2983)
                                                                           at android.app.ActivityThread.access$1800(ActivityThread.java:177)
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1526)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                           at android.os.Looper.loop(Looper.java:145)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:5951)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at java.lang.reflect.Method.invoke(Method.java:372)
                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
                                                                        Caused by: java.lang.ClassNotFoundException: Didn't find class "io.realm.internal.network.NetworkStateReceiver" on path: DexPathList[[zip file "/data/app/com.xyz.mobileapp-2/base.apk"],nativeLibraryDirectories=[/data/app/com.xyz.xyz-2/lib/arm, /vendor/lib, /system/lib]]
                                                                           at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                                                                           at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
                                                                           at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
                                                                           at android.app.ActivityThread.handleReceiver(ActivityThread.java:2978)
                                                                           at android.app.ActivityThread.access$1800(ActivityThread.java:177) 
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1526) 
                                                                           at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                           at android.os.Looper.loop(Looper.java:145) 
                                                                           at android.app.ActivityThread.main(ActivityThread.java:5951) 
                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                           at java.lang.reflect.Method.invoke(Method.java:372) 
                                                                           at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400) 
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195) 
                                                                       	Suppressed: java.lang.ClassNotFoundException: io.realm.internal.network.NetworkStateReceiver
                                                                           at java.lang.Class.classForName(Native Method)
                                                                           at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
                                                                           at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
                                                                           at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
                                                                           		... 11 more
                                                                        Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available
```

**Comment by christholey** [comment]
The same here. After a fresh install (Run...) via Android Studio the error occurs at app's startup. 
But it doesn't when I (re)start the app from the Android launcher.

**Comment by Zhuinden** [comment]
Duplicate of https://github.com/realm/realm-java/issues/3505

Will be fixed in 2.0.1

**Comment by phearme** [comment]
+1

**Comment by RJFares** [comment]
+1

**Comment by zaki50** [comment]
I'll close this as a duplicate.
We'll keep track this in #3505.

We treat this as a critical issue.
We'll release 2.0.1 ASAP.

**Comment by scijoker** [comment]
+1

**Comment by ahmadalibaloch** [comment]
Where is the Realm team sleeping. Recently the have been boasting to release new features and this fatal bug is not resolved after weeks 👎

**Comment by pc-bszabo** [comment]
Quick and dirty solution : put this into your launcher activity OR Application=>

try {
getApplicationContext().getPackageManager().setComponentEnabledSetting(new ComponentName("your.package.name","io.realm.internal.network.NetworkStateReceiver"), PackageManager.COMPONENT_ENABLED_STATE_DISABLED, PackageManager.DONT_KILL_APP);
} catch (Exception e) {}

**Comment by Zhuinden** [comment]
Fixed in 2.0.1 release
